consumer.go

```diff
@@ -454,7 +454,15 @@ func (c *Consumer) rebalance() (map[string][]int32, error) {
 	sarama.Logger.Printf("cluster/consumer %s rebalance\n", c.memberID)
 
 	if err := c.client.RefreshMetadata(); err != nil {
-		return nil, err
+		switch err {
```

Comment:
I think this can be expressed a bit simpler:

```go
err := c.client.RefreshMetadata()
if err = sarama.ErrTopicAuthorizationFailed {
  err = c.client.RefreshMetadata(c.coreTopics...)
}
return err
```